### ISSUE TITLE
loading pretrained model

### ISSUE THREAD

**Issue by hannesrosenbusch** [open-mmlab/mmfashion#4]
Thank you for these awesome resources!
I downloaded the 'ResNet-50 | Global Pooling' .pth file from your model zoo to my_path.
I assume I could load the model like this:
model = ???
model.load_state_dict(torch.load(my_path, map_location='cpu')

However I do not know how to initialize the ResNet model structure (see the "???").
I failed with using:
import torchvision.models as models
model = models.resnet50(pretrained=False)

Could you specify how to load the model? That would help me a lot.

**Comment by veralauee** [comment]
When testing, you can load the model like this:
python tools/test_predictor.py \
    --config configs/attribute_predict/roi_predictor_vgg_attr.py \
    --checkpoint [Pretrained model here]

For initialization,  
python tools/train_predictor.py \
    --config [your config file] \
    --resume_from [Pretrained model here]

**Comment by hannesrosenbusch** [comment]
Thank you for your reply. I would like to load the model without having to rely on nested mmfashion helper functions (they are difficult to follow/debug for pytorch beginners). Instead I tried the code below. It seems to work, but the model always predicts very extreme numbers and after applying softmax the top5 label predictions are always the same (regardless of the input image). Do you know what is going on? Am I maybe using the wrong network structure?

#CODE
import numpy as np
import torch
import torchvision

path = MY_PATH

#i found this network structure somewhere but cannot find it again in the mmfashion files
class ResNet(torchvision.models.resnet.ResNet):
    def __init__(self, *args, **kwargs):
        super(ResNet, self).__init__(*args, **kwargs) 

    def forward(self, x):
        x = self.conv1(x)
        x = self.bn1(x)
        x = self.relu(x)
        x = self.maxpool(x)

        x = self.layer1(x)
        x = self.layer2(x)
        x = self.layer3(x)
        x = self.layer4(x)

        x = self.avgpool(x)
        x = x.view(int(x.size(0)), -1)  # << This fix again
        x = self.fc(x)
        return x
model = ResNet(torchvision.models.resnet.Bottleneck, [3, 4, 6, 3])

model.load_state_dict(torch.load(path, map_location='cpu')['state_dict'], strict = False)
model.eval()

#load my own fashion images
inp = np.load('preprocessed_data.npy')

#change to channels first
inp = np.moveaxis(inp, 3,1)

#convert to double to prevent errors
inp = torch.from_numpy(inp).type('torch.DoubleTensor')
model = model.double()

#predict example image
pred = model(inp[108].unsqueeze(0))

#apply softmax as done in mmfashion issue
probs = torch.nn.functional.softmax(pred[0], dim=0)
top5 = torch.topk(probs, 5)
attr_cloth_file = open('C:/Users/rosenbus/surfdrive/pythanalytics/fashion labeling/list_attr_cloth.txt').readlines()
attr_idx2name = {}
for i, line in enumerate(attr_cloth_file[2:]):
  attr_idx2name[i] = line.strip('\n').split()[0]
for (i, v) in zip(top5[1], top5[0]):
  print(f"Label: {attr_idx2name[i.item()]}, Probs: {v.item()}.")
  
#output:
#Label: monroe, Probs: 1.0.
#Label: regime, Probs: 1.624687763424871e-92.
#Label: leopard, Probs: 2.7042922381237e-156.
#Label: hem, Probs: 3.910733863565495e-183.
#Label: pleated, Probs: 4.5121563790374856e-206.
